pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -44,13 +44,18 @@ const (
 )
 
 // TopologyAffinityError represents an resource alignment error
-type TopologyAffinityError struct{}
+type TopologyAffinityError struct {
+	Msg string
+}
 
-func (e TopologyAffinityError) Error() string {
+func (e *TopologyAffinityError) Error() string {
+	if e.Msg != "" {
+		return e.Msg
+	}
```

Comment:
I think we better add a new error. Repurposing this one has subtle implications as the error string made its way to the users and it's highly likely someone is already making logic on it. Which is a problem on its own. So let's start with a new error path and let's see later if it makes sense to merge with the old one.